public/redux/reducers/ad.ts

```diff
@@ -423,28 +423,29 @@ export const updateDetector = (
   detectorId,
 });
 
-export const deleteDetector = (detectorId: string): APIAction => ({
+export const deleteDetector = (detectorId: string, dataSourceId: string): APIAction => ({
   type: DELETE_DETECTOR,
   request: (client: HttpSetup) =>
-    client.delete(`..${AD_NODE_API.DETECTOR}/${detectorId}`),
+    client.delete(`..${AD_NODE_API.DETECTOR}/${detectorId}/${dataSourceId}`),
   detectorId,
 });
 
-export const startDetector = (detectorId: string): APIAction => ({
+export const startDetector = (detectorId: string, dataSourceId: string): APIAction => ({
   type: START_DETECTOR,
   request: (client: HttpSetup) =>
-    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/start`),
+    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/${dataSourceId}/start`),
   detectorId,
 });
 
 export const startHistoricalDetector = (
   detectorId: string,
+  dataSourceId: string,
   startTime: number,
   endTime: number
 ): APIAction => ({
   type: START_HISTORICAL_DETECTOR,
   request: (client: HttpSetup) =>
-    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/start`, {
+    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/${dataSourceId}/start`, {
```

Comment:
Same deal as `startDetector`

Suggested a change below:
```suggestion
    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}${dataSourceId ? '/' + dataSourceId : ''}/start`, {
```